pkg/daemon/upgrade_monitor_test.go

```diff
@@ -0,0 +1,175 @@
+package daemon
+
+import (
+	"context"
+	"testing"
+
+	apicfgv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/machine-config-operator/pkg/upgrademonitor"
+
+	"github.com/openshift/api/machineconfiguration/v1alpha1"
+	"github.com/openshift/client-go/machineconfiguration/clientset/versioned/fake"
+	informers "github.com/openshift/client-go/machineconfiguration/informers/externalversions"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeinformers "k8s.io/client-go/informers"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+type upgradeMonitorTestCase struct {
+	name               string
+	err                bool
+	parentCondition    *upgrademonitor.Condition
+	childCondition     *upgrademonitor.Condition
+	parentStatus       metav1.ConditionStatus
+	childStatus        metav1.ConditionStatus
+	expectedConditions []metav1.Condition
+}
+
+func TestUpgradeMonitor(t *testing.T) {
+	testCases := []upgradeMonitorTestCase{
+		{
+			name: "testUpdated",
+			err:  false,
+			parentCondition: &upgrademonitor.Condition{
+				State:   v1alpha1.MachineConfigNodeUpdated,
+				Reason:  "Updated",
+				Message: "Node Updated",
+			},
+			childCondition: nil,
+			parentStatus:   metav1.ConditionTrue,
+			childStatus:    metav1.ConditionFalse,
+			expectedConditions: []metav1.Condition{
+				{
+					Type:               string(v1alpha1.MachineConfigNodeUpdated),
+					Message:            "Node Updated",
+					Reason:             "Updated",
+					LastTransitionTime: metav1.Now(),
+					Status:             metav1.ConditionTrue,
+				},
+			},
+		},
+		{
+			name: "testUpdating",
+			err:  false,
+			parentCondition: &upgrademonitor.Condition{
+				State:   v1alpha1.MachineConfigNodeUpdateExecuted,
+				Reason:  "Updating",
+				Message: "Node Updating",
+			},
+			childCondition: &upgrademonitor.Condition{
+				State:   v1alpha1.MachineConfigNodeUpdateFilesAndOS,
+				Reason:  "FilesAndOS",
+				Message: "Applied Files and OS",
+			},
+			parentStatus: metav1.ConditionUnknown,
+			childStatus:  metav1.ConditionTrue,
+			expectedConditions: []metav1.Condition{
+				{
+					Type:               string(v1alpha1.MachineConfigNodeUpdateExecuted),
+					Message:            "Node Updating",
+					Reason:             "Updating",
+					LastTransitionTime: metav1.Now(),
+					Status:             metav1.ConditionUnknown,
+				},
+				{
+					Type:               string(v1alpha1.MachineConfigNodeUpdateFilesAndOS),
+					Message:            "Applied new Files and OS",
+					Reason:             "FilesAndOS",
+					LastTransitionTime: metav1.Now(),
+					Status:             metav1.ConditionTrue,
+				},
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		// Wire up the mutex to each test case before executing so they don't stomp
+		// on each other.
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
praise: Nice use of `t.Parallel()`!  It really helps to keep the test suite running as fast as possible which is important when one is making use of FakeClients.